`simple-env` has compilation error on TypeScript ^4.2.2

## Pre-requisites:
<!-- Add a check to each item below by replacing "[ ]" with "[x]" (no spaces inside the brackets, lowercase x) -->
- [x] Yes, I looked through both [open and closed issues](../../../issues?utf8=✓&q=is%3Aissue) looking for what I needed
- [x] No, I did not find what I was looking for

## Description
<!-- Enter a description of the issue/enhancement below -->
``` shell
node_modules/@americanairlines/simple-env/dist/types/Options.d.ts:7:18 - error TS2430: Interface 'InternalOptions' incorrectly extends interface 'Required<Options<DefaultEnvVars, DefaultEnvVars>>'.
  Types of property 'optional' are incompatible.
    Property '""' is missing in type 'DefaultEnvVars' but required in type 'Record<"", string>'.

7 export interface InternalOptions extends Required<Options> {
```

## Expected Behavior
<!-- What should have happened? -->
The dependency should build on all current versions of TypeScript

## Actual Behavior
<!-- What actually happened? -->
Versions ^4.1.5 cause a compilation error

## Repro Steps
<!-- Be as descriptive as possible -->
`npm i typescript@4.2.2 && npm run build`

--- a/jest.config.js
+++ b/jest.config.js
@@ -1,7 +1,7 @@
 module.exports = {
   globals: {
     'ts-jest': {
-      tsconfig: 'src/tsconfig.json',
+      tsconfig: 'tsconfig.json',
     },
   },
   moduleFileExtensions: ['js', 'json', 'ts'],

--- a/package.json
+++ b/package.json
@@ -14,7 +14,7 @@
   "scripts": {
     "test": "jest",
     "test:coverage": "npm run test -- --coverage",
-    "build": "tsc -p src/ --pretty",
+    "build": "tsc -p tsconfig.build.json --pretty",
     "build:watch": "npm run build -- -w",
     "start": "node dist/index.js",
     "start:watch": "nodemon -w dist -e js dist/index.js",
@@ -22,7 +22,8 @@
     "lint": "eslint \"src/**/*.ts\" --max-warnings 0",
     "lint:fix": "npm run lint -- --fix",
     "checkstyle": "prettier -l \"src/**/*.ts\"",
-    "checkstyle:fix": "npm run checkstyle -- --write"
+    "checkstyle:fix": "npm run checkstyle -- --write",
+    "tsc": "tsc -p tsconfig.build.json --noEmit --incremental false"
   },
   "repository": {
     "type": "git",
@@ -37,6 +38,7 @@
   "devDependencies": {
     "@typescript-eslint/eslint-plugin": "^4.8.1",
     "@typescript-eslint/parser": "^4.8.1",
+    "chalk": "^4.1.0",
     "eslint": "^7.13.0",
     "eslint-config-airbnb-base": "^14.2.1",
     "eslint-plugin-import": "^2.22.1",
@@ -47,6 +49,6 @@
     "prettier": "^2.1.2",
     "ts-jest": "^26.4.4",
     "ts-node": "^9.0.0",
-    "typescript": "^4.1.2"
+    "typescript": "^4.2.2"
   }
 }

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "@americanairlines/simple-env",
-  "version": "1.0.3",
+  "version": "1.0.4",
   "description": "An intuitive, strongly typed, and scalable way to retrieve environment variables",
   "keywords": [
     "env",

--- a/src/test/compilation.test.ts
+++ b/src/test/compilation.test.ts
@@ -2,25 +2,42 @@ import 'jest';
 import fs from 'fs';
 import path from 'path';
 import { exec } from 'child_process';
+import chalk from 'chalk';
 
 jest.setTimeout(10000);
 
-const shouldNotCompile = (filename: string) => async () => {
+const shouldNotCompile = (filename: string, tsErrors: string[]) => async () => {
   const fixturePath = path.join(process.cwd(), 'src/test/fixtures', filename);
   expect(fs.statSync(fixturePath).isFile()).toBe(true);
 
-  const child = exec(`npx tsc --noEmit --strict --lib es2019 ${fixturePath}`);
+  await new Promise<void>((resolve, reject) => {
+    exec(`npx tsc --noEmit --strict --lib es2019 ${fixturePath}`, (err, stdout) => {
+      expect(err?.code).toBeGreaterThan(0);
+      const foundTsErrors = stdout.match(/error TS\d+:/g)?.map((found) => found.replace('error ', '').replace(':', '')) ?? [];
 
-  await new Promise<void>((resolve) => {
-    child.on('exit', (code) => {
-      expect(code).toBeGreaterThan(0);
-      resolve();
+      try {
+        expect(foundTsErrors.sort()).toEqual(tsErrors.sort());
+        resolve();
+      } catch (jestError) {
+        const error = new Error(`
+${chalk.blueBright('==== Begin Typescript Compiler Output ====================')}
+
+${stdout.trim()}
+
+${chalk.blueBright('==== End Typescript Compiler Output ======================')}
+
+`);
+
+        error.stack = jestError.stack;
+
+        reject(error);
+      }
     });
   });
 };
 
 describe('compilation errors', () => {
-  it('missing key', shouldNotCompile('missingKey.ts'));
-  it('duplicate keys', shouldNotCompile('duplicateKeys.ts'));
-  it('optional key', shouldNotCompile('optionalKey.ts'));
+  it('missing key', shouldNotCompile('missingKey.ts', ['TS2339']));
+  it('duplicate keys', shouldNotCompile('duplicateKeys.ts', ['TS2322', 'TS2322', 'TS2322', 'TS2322']));
+  it('optional key', shouldNotCompile('optionalKey.ts', ['TS2532']));
 });

--- a/src/test/fixtures/duplicateKeys.ts
+++ b/src/test/fixtures/duplicateKeys.ts
@@ -2,5 +2,15 @@ import setEnv from '../..';
 
 setEnv({
   required: { unique1: '', duplicate: '' },
+  optional: { duplicate: '' },
+});
+
+setEnv({
+  required: { unique1: '', duplicate: '' },
   optional: { unique2: '', duplicate: '' },
+});
+
+setEnv({
+  required: { duplicate1: '', duplicate2: '' },
+  optional: { duplicate1: '', duplicate2: '' },
 });

--- a/src/types/Options.ts
+++ b/src/types/Options.ts
@@ -6,6 +6,6 @@ export interface Options<Required extends UndefinedEnvVars = DefaultEnvVars, Opt
   optional?: RemoveKeys<Optional, keyof Required>;
 }
 
-export interface InternalOptions extends Required<Options> {
+export interface InternalOptions extends Omit<Required<Options>, 'optional'> {
   optional: DefaultEnvVars;
 }

--- a/src/types/helpers.ts
+++ b/src/types/helpers.ts
@@ -1,5 +1,7 @@
 export type RemoveKeys<Obj, Keys> = keyof Obj extends Exclude<keyof Obj, Keys>
   ? Obj
-  : Record<Exclude<keyof Obj, Keys> extends never ? '' : Exclude<keyof Obj, Keys>, string>;
+  : Exclude<keyof Obj, Keys> extends never
+  ? Record<string, never>
+  : Record<Exclude<keyof Obj, Keys>, string>;
 
 export type SymbolWithDescription = symbol & { description: string };

--- a/tsconfig.build.json
+++ b/tsconfig.build.json
@@ -1,0 +1,9 @@
+{
+  "extends": "./tsconfig.json",
+  "compilerOptions": {
+    "incremental": true,
+    "noEmit": false
+  },
+  "include": ["src/**/*.ts"],
+  "exclude": ["node_modules", "src/test"]
+}

--- a/tsconfig.json
+++ b/tsconfig.json
@@ -3,7 +3,7 @@
     "module": "commonjs",
     "target": "es6",
     "moduleResolution": "node",
-    "outDir": "../dist",
+    "outDir": "dist",
     "esModuleInterop": true,
     "noImplicitAny": true,
     "noEmitOnError": true,
@@ -13,8 +13,8 @@
     "emitDecoratorMetadata": true,
     "experimentalDecorators": true,
     "declaration": true,
-    "strict": true
+    "strict": true,
+    "noEmit": true
   },
-  "include": ["./**/*.ts"],
-  "exclude": ["node_modules", "test"]
+  "include": ["src/**/*.ts"]
 }

